### PR TITLE
Update to sabre 3.0.8 for fseek fix

### PR DESCRIPTION
**Pull request by PVince81**
Fix for https://github.com/owncloud/core/issues/22476#issuecomment-196764305

@DeepDiver1975 @icewind1991 @rullzer @nickvergessen

@karlitschek backport to 9.0.1 ? This library update fixes issues with range requests when using external storage or federated shares (https). Range requests are used by the sync client when resuming aborted big downloads.

The bug itself is bad because it returns the **wrong** bytes when fseek fails.

### CODE REVIEW

File: sabre/dav/lib/CardDAV/Backend/PDO.php [hunk 1]
```diff
@@ -198,7 +198,7 @@ function deleteAddressBook($addressBookId) {
         $stmt = $this->pdo->prepare('DELETE FROM ' . $this->addressBooksTableName . ' WHERE id = ?');
         $stmt->execute([$addressBookId]);
 
-        $stmt = $this->pdo->prepare('DELETE FROM ' . $this->addressBookChangesTableName . ' WHERE id = ?');
+        $stmt = $this->pdo->prepare('DELETE FROM ' . $this->addressBookChangesTableName . ' WHERE addressbookid = ?');
```
Comment (PVince81) [inline sabre/dav/lib/CardDAV/Backend/PDO.php]: @DeepDiver1975 the update comes with this "bonus" fix, would that help with any other known issues ?

**Comment by rullzer** [comment]
I prefer it if we update the whole to sabre-3.0.8... So not just sabre/dav... 

`The zip release ships with sabre/vobject 3.5.0, sabre/http 4.2.1, sabre/event 2.0.2, sabre/uri 1.1.0 and sabre/xml 1.4.1.`

File: sabre/dav/lib/CardDAV/Backend/PDO.php (hunk 1 above)
Comment (DeepDiver1975) [inline sabre/dav/lib/CardDAV/Backend/PDO.php]: We do not use the PDO backend - THX

**Comment by DeepDiver1975** [comment]
> I prefer it if we update the whole to sabre-3.0.8... So not just sabre/dav...

full update on master - yes
backport TBH depends on the impact

**Comment by PVince81** [comment]
I first did the full update and there were changes in the XML lib and another one, then @nickvergessen suggested only updating sabre/dav.

I'm ok with doing a full update instead, then we can have a look at the code and think about what to backport.

**Comment by PVince81** [comment]
and here are the deps: https://github.com/owncloud/3rdparty/commit/1a3d64b7b2445b3df66d912751f48c3cec068734

**Comment by rullzer** [comment]
:+1:

**Comment by PVince81** [comment]
Let me do a core ticket for CI.

**Comment by PVince81** [comment]
core PR here https://github.com/owncloud/core/pull/23255

**Comment by PVince81** [comment]
All tests passed in core (except the usual unrelated SWIFT one...).

@DeepDiver1975 ok to merge ? (9.1)
Then we can discuss the backport part

**Comment by icewind1991** [comment]
:+1:

**Comment by PVince81** [comment]
I suggest we only backport the "sabre/dav" update to 9.0 to make sure we don't risk colateral regressions.
The library itself doesn't have a hard requirement on the latest sabre/xml so it should be fine.

@DeepDiver1975 @karlitschek ?

**Comment by DeepDiver1975** [comment]
@PVince81 agreed

**Comment by PVince81** [comment]
stable9: https://github.com/owncloud/3rdparty/pull/259

**Comment by karlitschek** [comment]
agreed! Thanks

**Pull request merged**